src/API/Trace/Propagation/TraceContextPropagator.php

```diff
@@ -22,20 +27,31 @@
  * traceparent header and relevant parts of the tracestate header containing
  * their proprietary information.
  */
-final class TraceContextPropagator implements API\TextMapPropagatorInterface
+final class TraceContextPropagator implements TextMapPropagatorInterface
 {
     public const TRACEPARENT = 'traceparent';
     public const TRACESTATE = 'tracestate';
     private const VERSION = '00'; // Currently, only '00' is supported
 
+    private static ?self $instance = null;
+
+    public static function getInstance(): self
+    {
+        if (null === self::$instance) {
+            self::$instance = new self();
+        }
+
+        return self::$instance;
+    }
+
     /** {@inheritdoc} */
-    public static function fields(): array
```

Comment:
What's the purpose of setting these as static functions?  Are you worried about errant constructors getting set here or something of the like?

Reply:
They were previously static functions since they handled injecting/extracting in a stateless manner. However, with the introduction of the multi processor, you could have more than one instantiation with different combination of propagators. For this reason i made the interface require them to be instance methods.